File: list/src/main/java/ru/igormayachenkov/list/DlgCommon.java
package ru.igormayachenkov.list;

////////////////////////////////////////////////////////////////////////////////////////////////////
// AlertDialog-BASED DIALOG
// FOR INFO SHOWING OR CONFIRMATION REQUEST

import android.content.Context;
import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;

public class DlgCommon {
    // EVENT INTERFACE
    public interface IEventListener {
        void onConfirmation();
    }

    // DATA
    IEventListener listener;
    AlertDialog alert;

    // CONSTRUCTOR
    public DlgCommon(Context context, int idTitle, int idMessage, IEventListener listener){
        this.listener = listener;

        // Create builder
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        if(idTitle!=0)   builder.setTitle(idTitle);
        if(idMessage!=0) builder.setMessage(idMessage);

        // Set up the buttons
        builder.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                // Raise event
                if (DlgCommon.this.listener != null)
                    DlgCommon.this.listener.onConfirmation();
            }
        });
        if(listener!=null)
            builder.setNegativeButton(android.R.string.cancel, null);

        // Create alert dialog
        alert = builder.create();
    }

    // SETTERS
    public  void setMessage(String message) {
        if (alert != null)
            alert.setMessage(message);
    }

    // SHOW
    public void show(){
        if(alert!=null)
            alert.show();
    }


}

File: list/src/main/java/ru/igormayachenkov/list/DlgCommon.kt
package ru.igormayachenkov.list

import android.R
import android.content.Context
import androidx.appcompat.app.AlertDialog

////////////////////////////////////////////////////////////////////////////////////////////////////
// AlertDialog-BASED DIALOG
// FOR INFO SHOWING OR CONFIRMATION REQUEST
class DlgCommon(
        context: Context,
        idTitle: Int,
        idMessage: Int,
        val callback: (()->Unit)?
        ) {
    var alert: AlertDialog

    // SETTERS
    fun setMessage(message: String) {
        alert.setMessage(message)
    }

    // SHOW
    fun show() {
        alert.show()
    }

    // CONSTRUCTOR
    init {

        // Create builder
        val builder = AlertDialog.Builder(context!!)
        if (idTitle != 0) builder.setTitle(idTitle)
        if (idMessage != 0) builder.setMessage(idMessage)

        // Set up the buttons
        builder.setPositiveButton(R.string.ok) { dialog, which -> // Raise event
            callback?.invoke()
        }
        if (callback != null) builder.setNegativeButton(R.string.cancel, null)

        // Create alert dialog
        alert = builder.create()
    }
}